### PR TITLE
Fix #2151 - Part 4 remove [Activate] support from controllers.

### LINKED ISSUE #2151: Naming inconsistency between [Activate] and @inject
Is there a reason why these are named differently?

 - In Razor it's currently `@inject`
 - In Mvc it's currently `[Activate]`

Why not `[Inject]` and `@inject`, or `[Activate]` and `@activate`?

### PR DESCRIPTION
**Pull request by rynowak**
This change completely removes [Activate]. In a controller, you should
constructor injection or [FromServices] to access services.

To access context items (ActionContext, ActionBindingContext, root
ViewDataDictionary) you should use the respective attribute class.

We'd like to consider streamlining this further in the future by getting
down to a single injectable context for controllers, but for now this will
have to do.

### CODE REVIEW

File: src/Microsoft.AspNet.Mvc.TagHelpers/ImageTagHelper.cs [hunk 1]
```diff
@@ -49,7 +49,6 @@ public ImageTagHelper(IHostingEnvironment hostingEnvironment, IMemoryCache cache
         [HtmlAttributeName(FileVersionAttributeName)]
         public bool FileVersion { get; set; }
 
-        [Activate]
```
Comment (dougbu) [inline src/Microsoft.AspNet.Mvc.TagHelpers/ImageTagHelper.cs]: Isn't this part of the other PR?

File: test/Microsoft.AspNet.Mvc.Core.Test/DefaultControllerFactoryTest.cs [hunk 2]
```diff
@@ -185,8 +184,10 @@ public void CreateController_ThrowsIfPropertyCannotBeActivated()
 
             // Act and Assert
             var exception = Assert.Throws<InvalidOperationException>(() => factory.CreateController(context));
-            Assert.Equal("The property 'Service' on controller '" + typeof(ControllerThatCannotBeActivated) +
-                        "' cannot be activated.", exception.Message);
+            Assert.Equal(
+                $"Unable to resolve service for type '{typeof(TestService)}' while attempting to activate " +
+                $"'{typeof(ControllerThatCannotBeActivated)}'.",
```
Comment (dougbu) [inline test/Microsoft.AspNet.Mvc.Core.Test/DefaultControllerFactoryTest.cs]: Minor icky-ness. Could you change to be explicit and use `.FullName` while here please?

**Comment by dougbu** [comment]
:shipit:

**Comment by pranavkm** [comment]
:shipit:

File: src/Microsoft.AspNet.Mvc.TagHelpers/ImageTagHelper.cs (hunk 1 above)
Comment (rynowak) [inline src/Microsoft.AspNet.Mvc.TagHelpers/ImageTagHelper.cs]: bingo.

File: test/Microsoft.AspNet.Mvc.Core.Test/DefaultControllerFactoryTest.cs (hunk 2 above)
Comment (rynowak) [inline test/Microsoft.AspNet.Mvc.Core.Test/DefaultControllerFactoryTest.cs]: Fixed locally.

**Pull request closed without merging**